[TH] Added `Writing`, `Article Styling Criteria`, `Beatmap Description`, `Category`, `Primary Metadata Source`, `TAG` (#6324)
ppy/osu-wiki: wiki/Article_styling_criteria/Writing/th.md

## การใช้อักษรตัวพิมพ์ใหญ่

### ชื่อบทความ

ชื่อบทความ เมื่อเขียนในย่อหน้าจะต้องไม่เป็นตัวพิมพ์ใหญ่ เว้นแต่จะเป็นลิงก์ที่ชี้ไปยังบทความหรือเป็นคำนามเฉพาะ (เช่น BanchoBot) ต่อไปนี้เป็นตัวอย่าง:

```markdown
หากคุณต้องการทราบข้อมูลเพิ่มเติมเกี่ยวกับการแชท โปรดดูที่ [Chat Console](/wiki/Chat_Console)
```

### ตัวดัดแปลงเกม

ตัวดัดแปลงเกมต้องมีคำว่า `mod` หลังชื่อ `mod` เพื่อลดความกำกวม ต้องสะกดตามที่เห็นด้านล่าง (ตัวพิมพ์ใหญ่และระยะห่างต้องตรงกัน):

- `Easy` หรือ `EZ`
- `No Fail` หรือ `NF`
- `Half Time` หรือ `HT`
- `Daycore` หรือ `DC`
- `Hard Rock` หรือ `HR`
- `Sudden Death` หรือ `SD`
- `Perfect` หรือ `PF`
- `Double Time` หรือ `DT`
- `Nightcore` หรือ `NC`
- `Fade In` หรือ `FI`
- `Hidden` หรือ `HD`
- `Flashlight` หรือ `FL`
- `Relax` หรือ `RL`
- `Autopilot` หรือ `AP`
- `Target Practice` หรือ `TP`
- `Spun Out` หรือ `SO`
- `1K`, `2K`, `3K`, `4K`, `5K`, `6K`, `7K`, `8K`, and `9K`
  - หากกล่าวถึงตัวดัดแปลงหลักโดยรวม ให้ใช้ `xK`.
- `Co-op` หรือ `CO`
- `Random` หรือ `RD`
- `Mirror` หรือ `MR`
- `Auto` หรือ `AT`
- `Cinema` หรือ `CM`
- `Touch Device` หรือ `TD`
- `ScoreV2`

---

> ตัวดัดแปลงเกมเหล่านี้ไม่ได้ใช้โดย osu!; อย่างไรก็ตาม หากจำเป็น จะต้องสะกดตามที่เห็นด้านล่าง (ตัวพิมพ์ใหญ่และตัวพิมพ์ควรตรงกัน):

Author: what ; for?

- `osu!`
- `osu!taiko`
- `osu!catch`
- `osu!mania`

การอ้างถึงชื่อโหมดเกมแบบเก่า (เช่น `Catch the Beat`, `Taiko` และ `Mania`) อาจทำได้หากพูดถึงชื่อเดิมของโหมดเกมดังกล่าว

## osu!

ชื่อของเกม osu! ต้องไม่เป็นตัวพิมพ์ใหญ่หรือตัวเอียง โอสุ! ตราสินค้าอย่างเป็นทางการจะต้องไม่ใช้ช่องว่างใด ๆ ตัวอย่างได้แก่

- `osu!academy`
- `osu!api`
- `osu!catch`
- `osu!direct`
- `osu!keyboard`
- `osu!mania`
- `osu!store`
- `osu!stream`
- `osu!supporter`
- `osu!tablet`
- `osu!taiko`
- `osu!talk`
- `osu!tourney`

ชื่อผู้ใช้ที่มี `osu!` ซึ่งเป็นส่วนหนึ่งของชื่อหัวข้อจะต้องเป็นตัวพิมพ์ใหญ่ ตัวอย่างได้แก่

- `osu! Alumni`
- `osu! Champion`

สำหรับเงื่อนไขอื่นๆ ทั้งหมด osu! ต้องถือว่าเป็น[คำนามที่มีคุณสมบัติ](https://en.wikipedia.org/wiki/Noun_adjunct) นี่หมายถึงการเพิ่มช่องว่างระหว่าง osu! และคำนามที่ปรับเปลี่ยน ตัวอย่างได้แก่

- `osu! tournaments`
- `osu! community`
- `osu! chat`
- `osu! client`
- `osu! wiki`

---

`osu!` ไม่ควรจบประโยค หากใช่ ให้ใช้จุดเต็ม (`.`) หรือเครื่องหมายคำถามเพื่อรับ `osu!.` หรือ `osu!?`

`osu!` ต้องไม่ตามด้วยเครื่องหมายอัศเจรีย์ (เช่น `osu!!`)   ประโยคจะต้องถูกเขียนใหม่เพื่อให้แน่ใจว่าสิ่งนี้จะไม่เกิดขึ้น

## คำศัพท์

คำเหล่านี้ต้องสะกดดังนี้ (การวรรคต้องตรงกัน):

> - `approach circle`
> - `game mode` (or `mode`)
> - `game mod` (or `mod`)
> - `gameplay`
> - `hit burst`
> - `hit circle`
> - `hitsound`
> - `in-game`
> - `playstyle`
> - `slider tick`
> - `slider ball`
> - `slider path`
> - `Kudosu` (เพื่อเป็นคำนามเฉพาะ)

Reviewer: ```suggestion
- `approach circle`
- `โหมดของเกม` (หรือ `โหมด`)
- `ม็อดของเกม` (หรือ `ม็อด`)
- `gameplay`
- `hit burst`
- `hit circle`
- `hitsound`
- `in-game`
- `รูปแบบการเล่น`
- `slider tick`
- `slider ball`
- `slider path`
- `Kudosu` (ให้ใช้เป็นเป็นคำนามเฉพาะ)
```

Author: uh maybe in-game can changed to ในเกม ?

---

บางคำมีรูปแบบต่างๆ ต้องใช้การสะกดคำที่ต้องการและมีดังนี้:

- `creator` หรือ `mapper` แทน `beatmapper`.
- `mapped` แทน `beatmapped`.

> - `BN` หรือ `Beatmap Nominators` เมื่อพูดถึง *ผู้เสนอบีทแมป*.

Reviewer: ```suggestion
- `BN` หรือ `Beatmap Nominators` เมื่อพูดถึง *ผู้เสนอบีทแมพ*.
```

- `sign in` แทน `log in`, เว้นแต่ชื่อของปุ่มหรือลิงก์จะระบุเป็นอย่างอื่น
- `sign out` แทน `log out`, เว้นแต่ชื่อของปุ่มหรือลิงก์จะระบุเป็นอย่างอื่น
- `register` แทน `sign up`, เว้นแต่ชื่อของปุ่มหรือลิงก์จะระบุเป็นอย่างอื่น

> ### บทความการแข่งขัน

Reviewer: ```suggestion
### บทความทัวร์นาเมนต์
```

> - `register` แทน `sign up`, เว้นแต่ชื่อของปุ่มหรือลิงก์จะระบุเป็นอย่างอื่น

Reviewer: @TicClick Sorry for the tag, but should this part be localized to Thai also?

Reviewer 2: (please don't apologize in vain)
which part? I think that English articles suffer the most from inconsistent terms (especially terms that are inconsistent with these on the website), and this is not required for Thai. use whatever is widely used out there in the community

Reviewer: Alright, Thanks

คำเหล่านี้ต้องสะกดดังนี้ (การวรรคต้องตรงกัน):

- `NoMods`, `NoMod`, หรือ `NM`
- `FreeMods`, `FreeMod`, หรือ `FM`
- `Tiebreaker`

## ตัวย่อ, คำย่อ และอักษรย่อ

ตัวย่อ, คำย่อ และอักษรย่อ ต้องมีความหมายเขียนออกมาเมื่อเกิดขึ้นครั้งแรก ครั้งอื่น ๆ เป็นทางเลือก แต่จะทำเมื่อจำเป็นเท่านั้น ต่อไปนี้เป็นตัวอย่าง:

```markdown
The NC (Nightcore) mod is similar to the DT (Double Time) mod because both NC and DT increase the speed of the music by 50%. However, NC will change the pitch of the music and add a clap and finish to the beat.
```

ตัวย่อ, คำย่อ และอักษรย่อ ต้องเป็นตัวพิมพ์ใหญ่ ยกเว้น ต่อไปนี้เป็นตัวอย่าง:

- `CS` สำหรับ `Circle Size`
- `AR` สำหรับ `Approach Rate`
- `DT` สำหรับ `Double Time`
- `SBS` สำหรับ `Storyboard Scripting`

---

ใช้ `e.g.` สำหรับ "ตัวอย่าง" และ `i.e.` สำหรับ "นั้นคือ".

---

ตัวย่อ, คำย่อ และอักษรย่อ ไม่ควรเป็นพหูพจน์ ต่อไปนี้เป็นตัวอย่าง:

```markdown

> Avoid:  BNs can nominate your beatmap.
> 
> Prefer: Beatmap Nominators can nominate your beatmap.

Reviewer: ```suggestion
หลีกเลี่ยง:  BNs can nominate your beatmap.

ให้ใช้: Beatmap Nominators can nominate your beatmap.
```

# การเขียน

*สำหรับมาตรฐาน การจัดรูปแบบ, ดูที่: [เกณฑ์รูปแบบบทความ/การจัดรูปแบบ](../Formatting)*

> *ข้อสังเกต: บทความนี้ใช้ [RFC 2119](https://tools.ietf.org/html/rfc2119) เพื่ออธิบายระดับความต้องการ*

Reviewer 2: this is also pointless (see my note on RFC somewhere below)

Author: uh english article included this so i put it in here. is it okay?

Reviewer 2: https://tools.ietf.org/html/rfc2119 explains how certain key words should be interpreted. these keywords are: "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL". if your translation uses neither of these, it might be a good idea to drop this line, because it makes no sense.

ทุกบทความที่เป็นภาษาอังกฤษต้องใช้ภาษาอังกฤษแบบปกติเท่านั้น

## ทะเบียนภาษา

วิกิมีการเขียนสองภาษา: เป็นกลางและไม่เป็นทางการ

การเขียนเป็นกลางเป็นค่าเริ่มต้นบนวิกิ

การเขียนอย่างไม่เป็นทางการนั้นพิเศษและมอบให้กับบทความจำนวนเล็กน้อย ตัวอย่างบางส่วนของการเขียนนี้ ได้แก่ [คำถามที่พบบ่อย](/wiki/FAQ) และ [ศูนย์ช่วยเหลือ](/wiki/Help_Centre)

สำหรับการเขียนทั้งสอง กฎทั้งหมดในส่วนนี้ของเกณฑ์ลักษณะบทความต้องปฏิบัติตาม จะมีการบันทึกข้อยกเว้นสำหรับการเขียนอย่างใดอย่างหนึ่ง

> ## ภาษาอังกฤษ

Reviewer 2: this looks irrelevant to Thai -- you are translating the article other Thai people will (ideally) use as a go-to guide

Author: ok

Author: oh yeah in thai there's no conflicting and same work stuff thing so should i put it out?

Reviewer 2: yes

```

## โหมดเกม

โหมดเกมจะต้องเขียนดังนี้:

- `osu!`
- `osu!taiko`
- `osu!catch`
- `osu!mania`

การอ้างถึงชื่อโหมดเกมแบบเก่า (เช่น `Catch the Beat`, `Taiko` และ `Mania`) อาจทำได้หากพูดถึงชื่อเดิมของโหมดเกมดังกล่าว

## osu!

ชื่อของเกม osu! ต้องไม่เป็นตัวพิมพ์ใหญ่หรือตัวเอียง โอสุ! ตราสินค้าอย่างเป็นทางการจะต้องไม่ใช้ช่องว่างใด ๆ ตัวอย่างได้แก่

- `osu!academy`
- `osu!api`
- `osu!catch`
- `osu!direct`
- `osu!keyboard`
- `osu!mania`
- `osu!store`
- `osu!stream`
- `osu!supporter`
- `osu!tablet`
- `osu!taiko`
- `osu!talk`
- `osu!tourney`

ชื่อผู้ใช้ที่มี `osu!` ซึ่งเป็นส่วนหนึ่งของชื่อหัวข้อจะต้องเป็นตัวพิมพ์ใหญ่ ตัวอย่างได้แก่

- `osu! Alumni`
- `osu! Champion`

สำหรับเงื่อนไขอื่นๆ ทั้งหมด osu! ต้องถือว่าเป็น[คำนามที่มีคุณสมบัติ](https://en.wikipedia.org/wiki/Noun_adjunct) นี่หมายถึงการเพิ่มช่องว่างระหว่าง osu! และคำนามที่ปรับเปลี่ยน ตัวอย่างได้แก่

- `osu! tournaments`
- `osu! community`
- `osu! chat`
- `osu! client`
- `osu! wiki`

---

`osu!` ไม่ควรจบประโยค หากใช่ ให้ใช้จุดเต็ม (`.`) หรือเครื่องหมายคำถามเพื่อรับ `osu!.` หรือ `osu!?`

`osu!` ต้องไม่ตามด้วยเครื่องหมายอัศเจรีย์ (เช่น `osu!!`)   ประโยคจะต้องถูกเขียนใหม่เพื่อให้แน่ใจว่าสิ่งนี้จะไม่เกิดขึ้น

## คำศัพท์

> คำเหล่านี้ต้องสะกดดังนี้ (การวรรคต้องตรงกัน):

Reviewer 2: this is also useless, unless you use words from the below list as-is even in translations. want to replace them with local equivalents?

typically, sections like these are used for words or collocations that may be misspelled in _your_ language, or have several spelling variations, and you need to pick one for consistency

Author: idk bout this though

Author: @Trigonoculus what should I do with this?
most of these words aren't in dictionary mostly

Reviewer: Some of these are better left untranslated (at least in my opinion). In our community, there are no "proper" translations to some of these. For example, Thai people called "approach circle" a "approach circle". However, some of these can definitely be transliterated to Thai (Well except approach circle cause that would be something like "อะโพรช เซอร์เคิล" which is probably harder to read than English).

Author: resolved
thanks @C0NS03L <3

```markdown
The NC (Nightcore) mod is similar to the DT (Double Time) mod because both NC and DT increase the speed of the music by 50%. However, NC will change the pitch of the music and add a clap and finish to the beat.
```

ตัวย่อ, คำย่อ และอักษรย่อ ต้องเป็นตัวพิมพ์ใหญ่ ยกเว้น ต่อไปนี้เป็นตัวอย่าง:

- `CS` สำหรับ `Circle Size`
- `AR` สำหรับ `Approach Rate`
- `DT` สำหรับ `Double Time`
- `SBS` สำหรับ `Storyboard Scripting`

---

ใช้ `e.g.` สำหรับ "ตัวอย่าง" และ `i.e.` สำหรับ "นั้นคือ".

---

ตัวย่อ, คำย่อ และอักษรย่อ ไม่ควรเป็นพหูพจน์ ต่อไปนี้เป็นตัวอย่าง:

```markdown
Avoid:  BNs can nominate your beatmap.

Prefer: Beatmap Nominators can nominate your beatmap.
```

## ทัศนคติ

`player`, `user`, `skinner`, `storyboarder`, และ `creator` ควรใช้เมื่อกล่าวถึงผู้อ่านหรือบุคคลอื่น `they`, `them`, `their`, และ `theirs` อาจใช้เมื่อจำเป็น

`you` และ `your` ควรหลีกเลี่ยง `I` จะต้องไม่ใช้ `we`, `he` หรือ `his`, และ `she` หรือ `her` จะต้องไม่ใช้ (ดูข้อกำหนดอื่น ๆ ด้านบน)

---

[บทความที่มีการเขียนอย่างไม่เป็นทางการ](#ทะเบียนภาษา) อาจละเว้นส่วนนี้ อย่างไรก็ตาม ควรหลีกเลี่ยง `I` ในย่อหน้า

## ไวยากรณ์

> บทความควรใช้ [ไวยากรณ์แบบอเมริกันที่ง่ายกว่า](https://www.thepunctuationguide.com/british-versus-american-style.html)

Reviewer 2: this.. well, you get what I mean. use local equivalents or, if there is no such issue in your language, just drop the section, ASC is one of the few articles that don't have to match the original 1:1

Author: um yeah I should use english one for that

Author: deleted

หากคุณต้องการทราบข้อมูลเพิ่มเติมเกี่ยวกับการแชท โปรดดูที่ [Chat Console](/wiki/Chat_Console)
```

### ตัวดัดแปลงเกม

ตัวดัดแปลงเกมต้องมีคำว่า `mod` หลังชื่อ `mod` เพื่อลดความกำกวม ต้องสะกดตามที่เห็นด้านล่าง (ตัวพิมพ์ใหญ่และระยะห่างต้องตรงกัน):

- `Easy` หรือ `EZ`
- `No Fail` หรือ `NF`
- `Half Time` หรือ `HT`
- `Daycore` หรือ `DC`
- `Hard Rock` หรือ `HR`
- `Sudden Death` หรือ `SD`
- `Perfect` หรือ `PF`
- `Double Time` หรือ `DT`
- `Nightcore` หรือ `NC`
- `Fade In` หรือ `FI`
- `Hidden` หรือ `HD`
- `Flashlight` หรือ `FL`
- `Relax` หรือ `RL`
- `Autopilot` หรือ `AP`
- `Target Practice` หรือ `TP`
- `Spun Out` หรือ `SO`
- `1K`, `2K`, `3K`, `4K`, `5K`, `6K`, `7K`, `8K`, and `9K`
  - หากกล่าวถึงตัวดัดแปลงหลักโดยรวม ให้ใช้ `xK`.
- `Co-op` หรือ `CO`
- `Random` หรือ `RD`
- `Mirror` หรือ `MR`
- `Auto` หรือ `AT`
- `Cinema` หรือ `CM`
- `Touch Device` หรือ `TD`
- `ScoreV2`

---

ตัวดัดแปลงเกมเหล่านี้ไม่ได้ใช้โดย osu!; อย่างไรก็ตาม หากจำเป็น จะต้องสะกดตามที่เห็นด้านล่าง (ตัวพิมพ์ใหญ่และตัวพิมพ์ควรตรงกัน):

- `Fade Out`
- `No Video`
- `10K`
- `Taiko`

---

เมื่อเขียนตัวดัดแปลงเกมสำหรับบทความการแข่งขัน พวกเขาต้องใช้ `upper camel case` แทน (ไม่ต้องเว้นวรรคและเก็บปลอกตัวอักษรตามที่เห็นด้านบน)

> <!-- I don't know what upper camel case is..-->

Author: i should skip it? bc i have no idea what even camel case look like lol

> <!-- I don't know what upper camel case is..-->

Author: maybe i should use คาเมลเคส since it's word adopt from english.

```markdown
In the osu! game mode, beatmaps are composed of three different gameplay elements: hit circles, sliders, and spinners.
```

### ชื่อภาษา

ชื่อภาษาต้องเป็นตัวพิมพ์ใหญ่ ต่อไปนี้เป็นตัวอย่าง:

```markdown
The `#spanish` chat channel is for those who speak Spanish.
```

### คำนามที่เหมาะสม

คำนามต้องเป็นตัวพิมพ์ใหญ่ ต่อไปนี้เป็นตัวอย่าง:

```markdown
Dean Herbert (also known as peppy) created osu! back in 2007.
```

### เครื่องหมายการค้า

*สำหรับกฎเกี่ยวกับ osu! โปรดดู: [osu!](#osu!)*

เครื่องหมายการค้าต่อไปนี้ต้องสะกดดังนี้ (ตัวพิมพ์ต้องตรงกัน):

- `Discord`
- `Facebook`
- `GitHub`
- `Google`
- `Reddit`
- `Skype`
- `Twitch`
- `Twitter`
- `YouTube`

เครื่องหมายการค้าต้องไม่ตามด้วยเครื่องหมายการค้าหรือเครื่องหมายการค้าจดทะเบียน

## วันและเวลา

### การจัดรูปแบบวันที่

รูปแบบที่พบใน [วิกิพีเดีย:คู่มือรูปแบบ/วันที่และตัวเลข § รูปแบบ](https://en.wikipedia.org/wiki/Wikipedia:Manual_of_Style/Dates_and_numbers#Formats) เป็นรูปแบบวันที่ที่ยอมรับทั้งหมด การจัดรูปแบบวันที่ควรสอดคล้องกันตลอดทั้งบทความ

รูปแบบวันที่ `ปปปป-ดด-วว` ควรใช้ในตารางเท่านั้น

### การจัดรูปแบบเวลา

การจัดรูปแบบเวลาต้องทำโดยใช้รูปแบบต่อไปนี้:

```markdown
ชั่วโมงเลข 2 ตัว:นาทีเลข 2 ตัว เขตเวลา
```

> เวลาเขียนในรูปแบบ 24 ชั่วโมง และตัวเลขเดี่ยวใดๆ จะต้องนำหน้าด้วยศูนย์ เขตเวลาจะต้องเขียนทันทีหลังจากเวลานั้น และควรอยู่ใน UTC+0 สำหรับเหตุการณ์ทั่วโลก หรือในเขตเวลาของกิจกรรมสำหรับกิจกรรมที่มีขนาดเล็กกว่า ใช้ `UTC`  แทน `GMT`

Reviewer 3: ```suggestion
เวลาเขียนในรูปแบบ 24 ชั่วโมง และตัวเลขเดี่ยวใดๆ จะต้องนำหน้าด้วยศูนย์ เขตเวลาจะต้องเขียนทันทีหลังจากเวลานั้น และควรอยู่ใน UTC+0 สำหรับเหตุการณ์ทั่วโลก หรือในเขตเวลาของกิจกรรมสำหรับกิจกรรมที่มีขนาดเล็กกว่า ใช้ `UTC` แทน `GMT`
```

ตัวอย่างที่ไม่ดี:

```markdown
3:30 PM UTC
22:30 (UTC+7)
11:30
```

ตัวอย่างที่ดี:

```markdown
15:30 UTC
22:30 UTC+7
11:30 UTC-4
```

`UTC` (ไม่มีออฟเซ็ต) ถูกบอกเป็นนัยว่าเป็น `UTC+0` อาจใช้อย่างใดอย่างหนึ่ง แต่การใช้งานต้องสอดคล้องกัน UTC ที่มีออฟเซ็ต 0 อย่างชัดเจนต้องใช้สัญลักษณ์บวก (`+`)

### การจัดรูปแบบวันที่และเวลา

เมื่อใช้ทั้งวันที่และเวลาร่วมกัน ต้องเขียนวันที่ก่อน ตามด้วยเวลา เวลาต้องอยู่ระหว่างวงเล็บ (`(` และ `)`)

ตัวอย่างที่ไม่ดี:

```markdown
October 25, 2016 at 11:45 UTC
October 25, 2016 11:45 UTC
```

ตัวอย่างที่ดี:

```markdown
October 25, 2016 (11:45 UTC)
```

## โหมดเกม

โหมดเกมจะต้องเขียนดังนี้:

- `osu!`
- `osu!taiko`
- `osu!catch`
- `osu!mania`

การอ้างถึงชื่อโหมดเกมแบบเก่า (เช่น `Catch the Beat`, `Taiko` และ `Mania`) อาจทำได้หากพูดถึงชื่อเดิมของโหมดเกมดังกล่าว

## osu!

> ชื่อของเกม osu! ต้องไม่เป็นตัวพิมพ์ใหญ่หรือตัวเอียง โอสุ! ตราสินค้าอย่างเป็นทางการจะต้องไม่ใช้ช่องว่างใด ๆ ตัวอย่าง ได้แก่

Reviewer 3: ```suggestion
ชื่อของเกม osu! ต้องไม่เป็นตัวพิมพ์ใหญ่หรือตัวเอียง โอสุ! ตราสินค้าอย่างเป็นทางการจะต้องไม่ใช้ช่องว่างใด ๆ ตัวอย่างได้แก่
```

- `osu!academy`
- `osu!api`
- `osu!catch`
- `osu!direct`
- `osu!keyboard`
- `osu!mania`
- `osu!store`
- `osu!stream`
- `osu!supporter`
- `osu!tablet`
- `osu!taiko`
- `osu!talk`
- `osu!tourney`

> ชื่อผู้ใช้ที่มี `osu!` ซึ่งเป็นส่วนหนึ่งของชื่อหัวข้อจะต้องเป็นตัวพิมพ์ใหญ่ ตัวอย่าง ได้แก่

Reviewer 3: ```suggestion
ชื่อผู้ใช้ที่มี `osu!` ซึ่งเป็นส่วนหนึ่งของชื่อหัวข้อจะต้องเป็นตัวพิมพ์ใหญ่ ตัวอย่างได้แก่
```

- `osu! Alumni`
- `osu! Champion`

สำหรับเงื่อนไขอื่นๆ ทั้งหมด osu! ต้องถือว่าเป็น[คำนามที่มีคุณสมบัติ](https://en.wikipedia.org/wiki/Noun_adjunct) นี่หมายถึงการเพิ่มช่องว่างระหว่าง osu! และคำนามที่ปรับเปลี่ยน ตัวอย่าง ได้แก่

- `osu! tournaments`
- `osu! community`
- `osu! chat`
- `osu! client`
- `osu! wiki`

---

`osu!` ไม่ควรจบประโยค หากใช่ ให้ใช้จุดเต็ม (`.`) หรือเครื่องหมายคำถามเพื่อรับ `osu!.` หรือ `osu!?`

> `osu!` ต้องไม่ตามด้วยเครื่องหมายอัศเจรีย์ (เช่น `osu!!`) ประโยคจะต้องเขียนใหม่เพื่อให้แน่ใจว่าสิ่งนี้จะไม่เกิดขึ้น

Reviewer 3: ```suggestion
`osu!` ต้องไม่ตามด้วยเครื่องหมายอัศเจรีย์ (เช่น `osu!!`) ประโยคจะต้องถูกเขียนใหม่เพื่อให้แน่ใจว่าสิ่งนี้จะไม่เกิดขึ้น
```

## คำศัพท์

> คำเหล่านี้ต้องสะกดดังนี้ (ระยะห่างต้องตรงกัน):

Reviewer 3: ```suggestion
คำเหล่านี้ต้องสะกดดังนี้ (การวรรคต้องตรงกัน):
```

- `approach circle`
- `game mode` (or `mode`)
- `game mod` (or `mod`)
- `gameplay`
- `hit burst`
- `hit circle`
- `hitsound`
- `in-game`
- `playstyle`
- `slider tick`
- `slider ball`
- `slider path`
- `Kudosu` (เพื่อเป็นคำนามเฉพาะ)

---

บางคำมีรูปแบบต่างๆ ต้องใช้การสะกดคำที่ต้องการและมีดังนี้:

- `creator` หรือ `mapper` แทน `beatmapper`.
- `mapped` แทน `beatmapped`.

> - `BN` หรือ `Beatmap Nominators` เมื่อพูดถึง *ผู้เสนอชื่อบีทแมป*.

Reviewer 3: ```suggestion
- `BN` หรือ `Beatmap Nominators` เมื่อพูดถึง *ผู้เสนอบีทแมป*.
```

- `sign in` แทน `log in`, เว้นแต่ชื่อของปุ่มหรือลิงก์จะระบุเป็นอย่างอื่น
- `sign out` แทน `log out`, เว้นแต่ชื่อของปุ่มหรือลิงก์จะระบุเป็นอย่างอื่น
- `register` แทน `sign up`, เว้นแต่ชื่อของปุ่มหรือลิงก์จะระบุเป็นอย่างอื่น

### บทความการแข่งขัน

คำเหล่านี้ต้องสะกดดังนี้ (ระยะห่างต้องตรงกัน):

> - `NoMods`, `NoMod`, or `NM`

Reviewer 3: ```suggestion
- `NoMods`, `NoMod`, หรือ `NM`
```

> - `FreeMods`, `FreeMod`, or `FM`

Reviewer 3: ```suggestion
- `FreeMods`, `FreeMod`, หรือ `FM`
```

- `Tiebreaker`

## ตัวย่อ, คำย่อ และอักษรย่อ

> ตัวย่อ, คำย่อ และอักษรย่อ ต้องมีความหมายเขียนออกมาเมื่อเกิดขึ้นครั้งแรก เหตุการณ์อื่นๆ เป็นทางเลือก แต่จะทำเมื่อจำเป็นเท่านั้น ต่อไปนี้เป็นตัวอย่าง:

Reviewer 3: ```suggestion
ตัวย่อ, คำย่อ และอักษรย่อ ต้องมีความหมายเขียนออกมาเมื่อเกิดขึ้นครั้งแรก ครั้งอื่น ๆ เป็นทางเลือก แต่จะทำเมื่อจำเป็นเท่านั้น ต่อไปนี้เป็นตัวอย่าง:
```

```markdown
The NC (Nightcore) mod is similar to the DT (Double Time) mod because both NC and DT increase the speed of the music by 50%. However, NC will change the pitch of the music and add a clap and finish to the beat.
```

ตัวย่อ, คำย่อ และอักษรย่อ ต้องเป็นตัวพิมพ์ใหญ่ ยกเว้น ต่อไปนี้เป็นตัวอย่าง:

- `CS` สำหรับ `Circle Size`
- `AR` สำหรับ `Approach Rate`
- `DT` สำหรับ `Double Time`
- `SBS` สำหรับ `Storyboard Scripting`

---

ใช้ `e.g.` สำหรับ "ตัวอย่าง" และ `i.e.` สำหรับ "นั้นคือ".

---

ตัวย่อ, คำย่อ และอักษรย่อ ไม่ควรเป็นพหูพจน์ ต่อไปนี้เป็นตัวอย่าง:

```markdown
Avoid:  BNs can nominate your beatmap.

Prefer: Beatmap Nominators can nominate your beatmap.
```

## ทัศนคติ

`player`, `user`, `skinner`, `storyboarder`, และ `creator` ควรใช้เมื่อกล่าวถึงผู้อ่านหรือบุคคลอื่น `they`, `them`, `their`, และ `theirs` อาจใช้เมื่อจำเป็น

> `you` และ `your` ควรหลีกเลี่ยง `I` จะต้องไม่ใช้ `we`, `he` หรือ `his`, และ `she` หรือ `her` จะต้องไม่ใช้ (ดูข้อกำหนดอื่นๆ ด้านบน)

Reviewer 3: ```suggestion
`you` และ `your` ควรหลีกเลี่ยง `I` จะต้องไม่ใช้ `we`, `he` หรือ `his`, และ `she` หรือ `her` จะต้องไม่ใช้ (ดูข้อกำหนดอื่น ๆ ด้านบน)
```

---

[บทความที่มีการลงทะเบียนอย่างไม่เป็นทางการ](#ทะเบียนภาษา) อาจละเว้นส่วนนี้ อย่างไรก็ตาม ควรหลีกเลี่ยง `I` ในย่อหน้า

## ไวยากรณ์

บทความควรใช้ [ไวยากรณ์แบบอเมริกันที่ง่ายกว่า](https://www.thepunctuationguide.com/british-versus-american-style.html)

### ศัพท์ย่อ

ไม่ควรใช้ศัพท์ย่อ

[บทความที่มีการลงทะเบียนอย่างไม่เป็นทางการ](#ทะเบียนภาษา) อาจใช้การย่อ; อย่างไรก็ตาม การใช้งานควรมีความสอดคล้องกันตลอดทั้งบทความ

### ตัวเลข

*สำหรับการจัดรูปแบบตัวเลข โปรดดูที่ [คู่มือรูปแบบการจัดรูปแบบตัวเลขของวิกิพีเดีย](https://en.wikipedia.org/wiki/Wikipedia:Manual_of_Style#Numbers)*

[บทความที่มีการลงทะเบียนอย่างไม่เป็นทางการ](#ทะเบียนภาษา) อาจละเว้นส่วนนี้ อย่างไรก็ตาม การใช้งานควรมีความสอดคล้องกันตลอดทั้งบทความ

### อัฒภาค

[บทความที่มีการลงทะเบียนอย่างไม่เป็นทางการ](#ทะเบียนภาษา) ควรหลีกเลี่ยงการใช้เครื่องหมายอัฒภาค (`;`)

### เครื่องหมายจุลภาค

ต้องใช้เครื่องหมายจุลภาคแบบอนุกรม หรือที่เรียกว่าเครื่องหมายจุลภาคของ Oxford หรือ Harvard

> ### Logical quotations

Reviewer 3: ```suggestion
### เครื่องหมายในการอ้างอิง
```
"Punctuation inside or outside"

Author: oh thanks
